packages/model/addon/-private/model.js

```diff
@@ -708,20 +708,24 @@ const Model = EmberObject.extend(DeprecatedEvented, {
 
     ```app/routes/model/delete.js
     import Route from '@ember/routing/route';
+    import { action } from '@ember/object';
 
-    export default Route.extend({
-      actions: {
-        softDelete() {
-          this.get('controller.model').deleteRecord();
-        },
-        confirm() {
-          this.get('controller.model').save();
-        },
-        undo() {
-          this.get('controller.model').rollbackAttributes();
-        }
+    export default class ModelDeleteRoute extends Route {
+      @action
+      softDelete() {
+        this.controller.get('model').deleteRecord();
       }
-    });
+
+      @action
+      confirm() {
+        this.controller.get('model').save();
+      }
+
+      @action
+      undo() {
+        this.controller.get('model').rollbackAttributes();
+      }
+    }
```

Comment:
Is this possible?  Perhaps we change this to a controller if you can't call an action on a route?

Reply:
I thought the original construct was a bit "odd" before I converted it. There must have been some reasoning for doing it that way instead of using a controller.

Reply:
I forgot.  You can call `sendAction` - first looking at the controller then up the active route path.  I'm sure new users would feel confused on how to call this action with the current guides.  I also don't think it is a prescribed path.  Up to you if you want to convert it!

Reply:
I'm going to convert to a controller.  `sendAction` is deprecated and you would need to use the `{{action}}` helper to use an action on the route